src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -66,6 +66,15 @@
  * r = 5; int t; //violation here
  * </pre>
  * <p>
+ * Note: variable references in try-with-resources are ignored by this Check.
```

Comment:
It is slightly misleading saying `variable references ... are ignored` as user would assume you mean all are ignored and there is no way to create violation on any. Your tests show violations are possible in specific situations.
Can we write some way here to denote that?
Something like `variable references with one assignment on a line in ...`?
Please also add a simple violation case to the code example.

Reply:
Do you mean something like:
`
Note: for resources declarations in try-with-resources variable references are not considered as statements and therefore not checked, but usual variable declarations are checked.  
`
I will add example to doc for couple of violation cases.

Reply:
> variable references are not ..., but usual variable declarations are ...

For me, my confusion is on what is the meaning between reference and declaration. The JLS calls it `VariableDeclaratorId` but I can't find variable reference listed to help guide me in your sentence. We try to use verbage closely related to the JLS.

Reply:
"variable reference" is called [VariableAccess](https://docs.oracle.com/javase/specs/jls/se9/html/jls-14.html#jls-VariableAccess) in JLS, we can use that. Maybe "variable reference" can be changed to "variable name", or we can use JLS terms (with or without links to JLS)

I updated docs, hope now it is more descriptive.

Btw, jenkins build failed for some infra issue.